src/lib/replication/log_store/home_raft_log_store.cpp

```diff
@@ -223,6 +221,18 @@ nuraft::ptr< std::vector< nuraft::ptr< nuraft::log_entry > > > HomeRaftLogStore:
     return out_vec;
 }
 
+nuraft::ptr< std::vector< nuraft::ptr< nuraft::log_entry > > >
+HomeRaftLogStore::log_entries_ext(ulong start, ulong end, int64_t batch_size_hint_in_bytes) {
+    // in nuraft , batch_size_hint_in_bytes < 0 indicats that follower is busy now and do not want to receive any more
+    // log entries ATM. here we just send one log entry if this happens which is helpful for nuobject case and no harm
+    // to other case.
+    if (batch_size_hint_in_bytes < 0) end = start + 1;
+
+    // for the case where batch_size_hint_in_bytes >= 0, we do not take any size check here for now.
+    // TODO: limit the size of the returned entries by batch_size_hint_in_bytes int the future if necessary
```

Comment:
@yamingk we discussed pressure_back , do you think we can use the batch_size_hint for this purpose?
i.e. follower can respond with `batch_size_hint` to leader through the response of appendEntries request.

Reply:
yes, looks like follower can set the next_batch_size_hint in append_entries_resp and leader will set it in its next batch size and consume it when next batch happens:
https://github.com/eBay/NuRaft/blob/eabdeeda538a27370943f79a2b08b5738b697ac3/src/handle_append_entries.cxx#L1069~1072